core/src/com/unciv/logic/city/IConstruction.kt

```diff
@@ -84,13 +84,24 @@ interface INonPerpetualConstruction : IConstruction, INamed, IHasUniques {
 
 
 class RejectionReasons: HashSet<RejectionReasonInstance>() {
-    
+
     fun add(rejectionReason: RejectionReason) = add(RejectionReasonInstance(rejectionReason))
-    
+
     fun contains(rejectionReason: RejectionReason) = any { it.rejectionReason == rejectionReason }
 
-    fun filterTechPolicyEraWonderRequirements(): List<RejectionReasonInstance> {
-        return filterNot { it.rejectionReason in techPolicyEraWonderRequirements }
+    fun isOKIgnoringRequirements(
+        ignoreTechPolicyEraWonderRequirements: Boolean = false,
+        ignoreResources: Boolean = false
+    ): Boolean {
+        if (!ignoreTechPolicyEraWonderRequirements && !ignoreResources) return isEmpty()
+        if (!ignoreTechPolicyEraWonderRequirements)
+            return none { it.rejectionReason != RejectionReason.ConsumesResources }
```

Comment:
"Glass half full" - yes, I went with the first variant that came to mind after inverting the original logic twice.

Reply:
...but `any` would be incorrect. `all` does it correctly - "all of them are stuff that we don't care about for the moment" instead of "there's none we _do_ care about - which is any in _not_ the list of don't cares". I'm almost 50:50 on what is more readable.

Reply:
And believe it or not I had that here and reverted. 'cuz the first test should be the most likely one for some picoseconds.